app/src/main/java/io/appium/uiautomator2/server/ServerConfig.java

```diff
@@ -16,10 +16,131 @@
+    private final static int MJPEG_SERVER_PORT =
+        getValueFromEnvOrDefault("MJPEG_SERVER_PORT", 7810);
+    // mjpeg fps should be between `1` and `60` (default: `10`)
+    private final static int MJPEG_SERVER_FRAMERATE =
+        getValueFromEnvOrDefault("MJPEG_SERVER_FRAMERATE", 10);
+    // mjpeg scale should be between `1` and `100` (default: `50`)
+    private final static int MJPEG_SCALING_FACTOR =
+        getValueFromEnvOrDefault("MJPEG_SCALING_FACTOR", 50);
+    // mjpeg quality is between 0 and 100 (default: `50`)
+    private final static int MJPEG_SERVER_SCREENSHOT_QUALITY =
+        getValueFromEnvOrDefault("MJPEG_SERVER_SCREENSHOT_QUALITY", 50);
+    // if filtering should be used in mjpeg resize operation (default: `false`)
+    private final static boolean MJPEG_BILINEAR_FILTERING =
+        Boolean.parseBoolean(System.getenv("MJPEG_BILINEAR_FILTERING"));
+
+    // In-memory overrides
+    private static Map<String, Object> overrides = new HashMap<>();
+
+    /* Helper Method(s) */
+
+    private static Integer getValueFromEnvOrDefault(String key, Integer defaultValue) {
+        String value = System.getenv(key);
+        if (value == null) {
+            return defaultValue;
+        }
+        return Integer.parseInt(value);
+    }
+
+    private static Object getValueFromOverridesOrDefault(String key, Object defaultValue) {
+        synchronized (overrides) {
+            if (overrides.containsKey(key)) {
+                return overrides.get(key);
+            }
+            return defaultValue;
+        }
+    }
+
+    private static void setOverridesValue(String key, Object value) {
+        synchronized (overrides) {
+            overrides.put(key, value);
+        }
+    }
+
+    /* Getter(s) */
 
     public static int getServerPort() {
-        return PORT;
+        return (int) getValueFromOverridesOrDefault(
+            ServerPort.SETTING_NAME,
+            SERVER_PORT);
+    }
+
+    public static int getMjpegServerPort() {
+        return (int) getValueFromOverridesOrDefault(
+            MjpegServerPort.SETTING_NAME,
+            MJPEG_SERVER_PORT);
+    }
+
+    public static int getMjpegServerFramerate() {
+        return (int) getValueFromOverridesOrDefault(
+            MjpegServerFramerate.SETTING_NAME,
+            MJPEG_SERVER_FRAMERATE);
+    }
+
+    public static int getMjpegServerScreenshotQuality() {
+        return (int) getValueFromOverridesOrDefault(
+            MjpegServerScreenshotQuality.SETTING_NAME,
+            MJPEG_SERVER_SCREENSHOT_QUALITY);
+    }
+
+    public static int getMjpegScalingFactor() {
+        return (int) getValueFromOverridesOrDefault(
+            MjpegScalingFactor.SETTING_NAME,
+            MJPEG_SCALING_FACTOR);
+    }
+
+    public static boolean isMjpegBilinearFiltering() {
+        return (boolean) getValueFromOverridesOrDefault(
+            MjpegBilinearFiltering.SETTING_NAME,
+            MJPEG_BILINEAR_FILTERING);
+    }
+
+    /* Setter(s) */
```

Comment:
I usually use regions notation to make it work better with IDEs. See https://blog.jetbrains.com/idea/2012/03/custom-code-folding-regions-in-intellij-idea-111/

Reply:
I just removed the comments since they really weren't adding a ton of value / clarification over the obvious.

Reply:
This will be resolved w/ the upcoming update(s)